### PR TITLE
200 202 stream resolving

### LINKED ISSUE #200: Op-less write to stream doesn't actually write data to stream
This Aqua code will return an empty stream
```scala
service Op("op"):
    noop()
    get_records(key: string) -> []string

func append_records(peer: string, stream: *[]string):
  on peer:
    stream <- Op.get_records(peer)

func retrieve_records(peer: string) -> [][]string:
    records: *[]string
    append_records(peer, records)
    <- records
```

Here's AIR generated by **Aqua Compiler 0.1.9-162**
```clojure
(xor
 (seq
  (seq
   (seq
    (seq
     (seq
      (call %init_peer_id% ("getDataSrv" "-relay-") [] -relay-)
      (call %init_peer_id% ("getDataSrv" "peer") [] peer)
     )
     (call -relay- ("op" "noop") [])
    )
    (xor
     (call peer ("op" "get_records") [peer] $stream)
     (seq
      (call -relay- ("op" "noop") [])
      (call %init_peer_id% ("errorHandlingSrv" "error") [%last_error% 1])
     )
    )
   )
   (call -relay- ("op" "noop") [])
  )
  (xor
   (call %init_peer_id% ("callbackSrv" "response") [$records])
   (call %init_peer_id% ("errorHandlingSrv" "error") [%last_error% 2])
  )
 )
 (call %init_peer_id% ("errorHandlingSrv" "error") [%last_error% 3])
)
```

### LINKED ISSUE #202: Invalid stream name scoping
The following Aqua code provides an invalid AIR
```
service StringOp("op"):
    array(s: string) -> []string

func use_name1() -> []string:
    results <- StringOp.array("hello")
    <- results

func use_name2() -> [][]string:
    results: *[]string
    results <- use_name1()
    results <- use_name1()
    results <- use_name1()
    <- results
```

AIR validation error
```
multiple variables found for name 'results' in data","instruction":"call %init_peer_id% ("op" "array") ["hello"] results"
```

AIR produced by **Aqua Compiler 0.1.9-162**:
```clojure
(xor
 (seq
  (seq
   (seq
    (seq
     (call %init_peer_id% ("getDataSrv" "-relay-") [] -relay-)
     (call %init_peer_id% ("op" "array") ["hello"] results)
    )
    (call %init_peer_id% ("op" "array") ["hello"] results)
   )
   (call %init_peer_id% ("op" "array") ["hello"] results)
  )
  (xor
   (call %init_peer_id% ("callbackSrv" "response") [results])
   (call %init_peer_id% ("errorHandlingSrv" "error") [%last_error% 1])
  )
 )
 (call %init_peer_id% ("errorHandlingSrv" "error") [%last_error% 2])
)
```

### PR DESCRIPTION
**Pull request by DieMyst**
fixes #200  
fixes #202

### CODE REVIEW

File: model/src/main/scala/aqua/model/func/FuncCallable.scala [hunk 1]
```diff
@@ -61,10 +77,11 @@ case class FuncCallable(
 
     // Substitute arguments (referenced by name and optional lambda expressions) with values
     // Also rename all renamed arguments in the body
-    val treeWithValues = body.rename(argsShouldRename).resolveValues(argsToData)
+    val treeWithValues = body.rename(argsShouldRename ++ streamToRename).resolveValues(argsToData)
 
     // Function body on its own defines some values; collect their names
-    val treeDefines = treeWithValues.definesVarNames.value -- call.exportTo.map(_.name)
```
Comment (alari) [inline model/src/main/scala/aqua/model/func/FuncCallable.scala:67]: Why `call.exportTo.map(_.name)` is removed?

File: model/src/main/scala/aqua/model/func/FuncCallable.scala [hunk 2]
```diff
@@ -129,9 +146,24 @@ case class FuncCallable(
...
-      .map { case ((_, resolvedExports), b) =>
+      .map { case ((_, resolvedExports), callableFuncBody) =>
         // If return value is affected by any of internal functions, resolve it
-        FuncOp(b) -> result.map(_.resolveWith(resolvedExports))
+        (for {
+          exp <- call.exportTo
+          res <- result
+          pair <- exp match {
+            // TODO check variable, not string
+            case Call.Export(name, StreamType(_)) if name == "-return-" =>
```
Comment (alari) [inline model/src/main/scala/aqua/model/func/FuncCallable.scala:156]: Where does this case come from?

File: model/src/main/scala/aqua/model/func/FuncCallable.scala [hunk 3]
```diff
@@ -129,9 +146,24 @@ case class FuncCallable(
...
+              // don't create identity for `-return-` stream
+              Some(FuncOp(callableFuncBody) -> result)
+            case Call.Export(name, StreamType(_)) =>
+              // path nested function results to a stream
+              Some(FuncOps.seq(FuncOp(callableFuncBody), FuncOps.identity(res, exp)) -> result)
```
Comment (alari) [inline model/src/main/scala/aqua/model/func/FuncCallable.scala:161]: Why not resolve the result with resolved exports?

File: model/src/main/scala/aqua/model/func/FuncCallable.scala (hunk 2 above)
Comment (DieMyst) [reply model/src/main/scala/aqua/model/func/FuncCallable.scala:156]: all will be wrapped with `identity result -return-`, if result is a stream

File: model/test-kit/src/test/scala/aqua/model/transform/TransformSpec.scala [hunk 4]
```diff
@@ -148,11 +148,11 @@ class TransformSpec extends AnyFlatSpec with Matchers {
           CallServiceRes(
             LiteralModel.quote("srv1"),
             "foo",
-            Call(Nil, Some(Call.Export("v", ScalarType.string))),
+            Call(Nil, Some(Call.Export("v0", ScalarType.string))),
```
Comment (alari) [inline model/test-kit/src/test/scala/aqua/model/transform/TransformSpec.scala:151]: Why did this change happen?

File: model/src/main/scala/aqua/model/func/FuncCallable.scala (hunk 3 above)
Comment (DieMyst) [reply model/src/main/scala/aqua/model/func/FuncCallable.scala:161]: it shouldn't be resolved if a result is a stream. Everything should be done before

File: model/src/main/scala/aqua/model/transform/ResolveFunc.scala [hunk 5]
```diff
@@ -12,7 +12,8 @@ case class ResolveFunc(
   callback: (String, Call) => FuncOp,
   respFuncName: String,
   wrapCallableName: String = "funcAround",
-  arrowCallbackPrefix: String = "init_peer_callable_"
+  arrowCallbackPrefix: String = "init_peer_callable_",
+  returnVar: String = "-return-"
```
Comment (alari) [inline model/src/main/scala/aqua/model/transform/ResolveFunc.scala:16]: Is there any reason to keep this? It should always be resolved & replaced internally, never goes to generated code

**Review by alari — APPROVED** [review]

**Pull request merged**